### PR TITLE
imx6ull-flashnor: improve build

### PR DESCRIPTION
**Pull request by badochov**
JIRA: RTOS-484

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imx6ull-flashnor/Makefile [hunk 1]
```diff
@@ -4,10 +4,18 @@
 # Copyright 2021 Phoenix Systems
 #
 
-NAME := imx6ull-flashnor
-LOCAL_SRCS := flashnor.c flashnor-ecspi.c flashnor-qspi.c flashnor-drv.c nor/nor.c
-LOCAL_HEADERS := flashnor-ecspi.h flashnor-qspi.h flashnor-drv.h nor/nor.h
+
+NAME := libflashnordrv-imx6ull
+LOCAL_SRCS := flashnor-ecspi.c flashnor-qspi.c flashnor-drv.c nor.c
+LOCAL_HEADERS := imx6ull-flashnor-ecspi.h imx6ull-flashnor-qspi.h imx6ull-flashnor-drv.h imx6ull-nor.h
```
Comment (nalajcie) [inline storage/imx6ull-flashnor/Makefile:10]: `DEP_LIBS` is not a valid parameter for `static-lib.mk`, maybe you meant `DEPS`? 

https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/makes/static-lib.mk#L10

Comment (nalajcie) [inline storage/imx6ull-flashnor/Makefile:10]: Are these all files external API?
Maybe put them as
`include/imx6ull-flashnor/ecspi.h` ... 

And use as `#include <imx6ull-flashnor/ecspi.h>`?

Edit:
As this lib is being used only inside the same dir (component) I would opt not to make it an explicit library (don't install headers, use objects directly - make would ensure they would be built only once and in correct order).

I'm thinking about something along the lines (not tested):

```make
IMX6ULL_FLASHNOR_COMMON_SRCS := flashnor-ecspi.c flashnor-qspi.c flashnor-drv.c nor.c

NAME := imx6ull-flashnor
LOCAL_SRCS := $(IMX6ULL_FLASHNOR_COMMON_SRCS) flashnor-srv.c
DEP_LIBS := libimx6ull-ecspi libimx6ull-qspi
LIBS := libmeterfs-storage libmeterfs libstorage libjffs2 libmtd
LOCAL_INSTALL_PATH := /sbin
include $(binary.mk)

NAME := flashnor-test
LOCAL_SRCS := $(IMX6ULL_FLASHNOR_COMMON_SRCS) flashnor-test.c
DEP_LIBS := libimx6ull-ecspi libimx6ull-qspi
include $(binary.mk)
```

File: storage/imx6ull-flashnor/Makefile [hunk 2]
```diff
@@ -4,10 +4,18 @@
...
 DEP_LIBS := libimx6ull-ecspi libimx6ull-qspi
+
+include $(static-lib.mk)
+
+NAME := imx6ull-flashnor
+LOCAL_SRCS := flashnor-srv.c
+LOCAL_HEADERS :=
```
Comment (nalajcie) [inline storage/imx6ull-flashnor/Makefile:17]: variables are automatically reset in `static-lib.mk` (https://github.com/phoenix-rtos/phoenix-rtos-build/blob/master/makes/static-lib.mk#L112), no need to explicitly empty them

File: storage/imx6ull-flashnor/Makefile [hunk 3]
```diff
@@ -16,6 +24,7 @@ include $(binary.mk)
 # NOR flash test program
 NAME := flashnor-test
 LOCAL_SRCS := flashnor-test.c
+DEP_LIBS := libflashnordrv-imx6ull libimx6ull-ecspi libimx6ull-qspi
 LIBS :=
```
Comment (nalajcie) [inline storage/imx6ull-flashnor/Makefile:28]: no need to reset

**Review by nalajcie — COMMENTED** [review]
pointed out some issues which might still be improved ;)

File: spi/imx6ull-ecspi/Makefile [hunk 4]
```diff
@@ -6,5 +6,5 @@
 
 NAME := libimx6ull-ecspi
 LOCAL_SRCS := libecspi.c
-LOCAL_HEADERS := ecspi.h
+LOCAL_HEADERS := imx6ull-ecspi.h
```
[the remark this answers is not in the document]
Comment (badochov) [reply spi/imx6ull-ecspi/Makefile:9]: When I try to put this header in the include directory I am not able to compile.
`phoenix-rtos-devices/spi/imx6ull-qspi`
```
#
# Makefile for Phoenix-RTOS imx6ull-qspi driver
#
# Copyright 2023 Phoenix Systems
#

NAME := libimx6ull-qspi
LOCAL_SRCS := qspi.c
include $(static-lib.mk)
```

Folder structure:
image

Error:
image


What am I doing wrong?

[the remark this answers is not in the document]
Comment (badochov) [reply spi/imx6ull-ecspi/Makefile:9]: When I try to put this header in the include directory I am not able to compile.
`phoenix-rtos-devices/spi/imx6ull-qspi/Makefile`
```
#
# Makefile for Phoenix-RTOS imx6ull-qspi driver
#
# Copyright 2023 Phoenix Systems
#

NAME := libimx6ull-qspi
LOCAL_SRCS := qspi.c
include $(static-lib.mk)
```

Folder structure:
image

Error:
image


What am I doing wrong?

[the remark this answers is not in the document]
Comment (nalajcie) [reply spi/imx6ull-ecspi/Makefile:9]: did You do `clean` build? If not, the autogenerated deps would still reference the old file name.

For me (on current master) change like that compiles without errors (after `clean`):
image

[the remark this answers is not in the document]
Comment (badochov) [reply spi/imx6ull-ecspi/Makefile:9]: ok, I have it I cahnged include in qspi.c to "include/imx6ull-qspi.h" it works with "imx6ull-qspi.h"

**Review by nalajcie — COMMENTED** [review]
one more thing and we're ready to merge

**Review by nalajcie — APPROVED** [review]

**Pull request merged**